sky/jobs/server/core.py

```diff
@@ -100,9 +101,33 @@ def launch(
 
     with rich_utils.safe_status(
             ux_utils.spinner_message('Initializing managed job')):
+
         for task_ in dag.tasks:
-            controller_utils.maybe_translate_local_file_mounts_and_sync_up(
-                task_, task_type='jobs')
+            if storage_lib.get_cached_enabled_storage_clouds_or_refresh():
+                controller_utils.maybe_translate_local_file_mounts_and_sync_up(
+                    task_, task_type='jobs')
+                first_hop_file_mounts = {}
+
+            else:
+                # We do not have any cloud storage available, so fall back to
+                # two-hop file_mount uploading.
+                # Note: we can't easily hack sync_storage_mounts() to upload
+                # directly to the controller, because the controller may not
+                # even be up yet.
+
+                if task_.storage_mounts:
+                    # Technically, we could convert COPY storage_mounts that
+                    # have a local source and do not specify `store`, but we
+                    # will not do that for now. Only plain file_mounts are
+                    # supported.
+                    raise exceptions.NotSupportedError(
+                        'Cloud-based file_mounts are specified, but no cloud '
+                        'storage is available. Please specify local '
+                        'file_mounts only.')
+
+                first_hop_file_mounts = (
```

Comment:
This variable name is a bit confusing. How about `user_file_mounts_to_controller`?